addons/cloud_storage/models/mail_mail.py

```diff
@@ -0,0 +1,49 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+
+from odoo import models, tools
+
+
+class MailMail(models.Model):
+    """Add the mass mailing campaign data to mail"""
+    _inherit = ['mail.mail']
+
+    def _prepare_outgoing_list(self, mail_server=False, recipients_follower_status=None):
+        outgoing_list = super()._prepare_outgoing_list(
+            mail_server=mail_server,
+            recipients_follower_status=recipients_follower_status
+        )
+
+        # Turn remaining attachments into links appended to an email if they are cloud attachments
+        for mail, outgoing in zip(self, outgoing_list):
+            cloud_attachments = mail.attachment_ids.filtered(
+                lambda attachment: attachment.type == 'cloud_storage'
+                    and attachment.res_model != 'mail.message'
```

Comment:
I think this condition is not useful.
Technically, now cloud stroage attachments won't be attached to `mail.message`. But even if we can attachment them. We should change the attachment to the `'mail.mail_attachment_links'` becasue otherwise the `raw` of these attachments will still be empty.